src/main/java/com/karankumar/bookproject/ui/shelf/BooksInShelfView.java

```diff
@@ -91,7 +92,7 @@ public BooksInShelfView(BookService bookService, PredefinedShelfService predefin
         this.bookGrid = new BookGrid(predefinedShelfUtils, customShelfUtils);
         this.bookFilters = new BookFilters();
 
-        this.whichShelf = new BookShelfComboBox(customShelfService);
+        this.whichShelf = new BookShelfComboBox(customShelfService, this);
```

Comment:
Isn't this creating a cyclic dependency? ShelfView has a combo box and combo box has the shelf?

Reply:
I didn't realize the bind() calls later in the code helped make what I wrote here unnecessary. I've removed this component in my latest code.